modules/gdscript/gdscript_editor.cpp

```diff
@@ -79,7 +79,8 @@ Ref<Script> GDScriptLanguage::make_template(const String &p_template, const Stri
 	type_hints = EDITOR_GET("text_editor/completion/add_type_hints");
 #endif
 	if (!type_hints) {
-		processed_template = processed_template.replace(": int", "")
+		processed_template = processed_template.replace("class_name _CLASS_\n", "")
```

Comment:
This will remove the class name for those templates where the name is required/desired.

https://github.com/godotengine/godot/blob/89850d553eeb259e208d0c577cd7bc1eabd3a90a/modules/gdscript/editor/script_templates/VisualShaderNodeCustom/basic.gd#L4-L5

https://github.com/godotengine/godot/blob/89850d553eeb259e208d0c577cd7bc1eabd3a90a/modules/gdscript/editor/script_templates/RichTextEffect/default.gd#L4-L5

Reply:
This actually does not remove the class name for these templates, since it is not the same string. Tested it.

Reply:
Those specific ones yes, but you can have custom templates, or we might add new ones

Reply:
Well, not here specifically, but in similar cases, if we or the user in a custom template omit/forget the prefix.